src/vivarium_census_prl_synth_pop/utilities.py

```diff
@@ -524,23 +524,29 @@ def copy_from_household_member(
         households = households.loc[
             households.index.difference(data_values.GQ_HOUSING_TYPE_MAP.keys())
         ]
-        simulants_and_household_members = pop["household_id"].map(households).dropna()
-        simulant_ids_to_copy = simulants_and_household_members.reset_index().apply(
-            lambda row: [
-                household
-                for household in row.loc["household_id"]
-                if household != row.loc["index"]
-            ],
-            axis=1,
-        )
-        simulant_ids_to_copy.index = simulants_and_household_members.index
-        seed = get_hash(randomness_stream._key(additional_key=col))
-        copy_ids = simulant_ids_to_copy.map(np.random.default_rng(seed).choice)
-        if col == "has_ssn":
-            pop.loc[copy_ids.index, copy_cols[col]] = copy_ids
-        else:
-            pop.loc[copy_ids.index, copy_cols[col]] = copy_ids.map(
-                pop.loc[copy_ids.index, col]
+        if not households.empty:
+            simulants_and_household_members = pop["household_id"].map(households).dropna()
+            # Note the following call results in a dataframe if 'simulants_and_household_members' is empty
+            # and is handled above
+            simulant_ids_to_copy = simulants_and_household_members.reset_index().apply(
+                lambda row: [
+                    household
+                    for household in row.loc["household_id"]
+                    if household != row.loc["index"]
+                ],
+                axis=1,
             )
+            simulant_ids_to_copy.index = simulants_and_household_members.index
+            seed = get_hash(randomness_stream._key(additional_key=col))
+            copy_ids = simulant_ids_to_copy.map(np.random.default_rng(seed).choice)
+            if col == "has_ssn":
+                pop.loc[copy_ids.index, copy_cols[col]] = copy_ids
+            else:
+                pop.loc[copy_ids.index, copy_cols[col]] = copy_ids.map(
+                    pop.loc[copy_ids.index, col]
+                )
+        else:
+            # Save as object type - current pandas defaults to dtype float with future warning
+            pop[copy_cols[col]] = pd.Series(np.nan, index=pop.index, dtype=object)
```

Comment:
So this else block is the new. I'm wondering if there's a good way to avoid all these indents. Not sure it's an improvement (i.e., take it or leave it imho), but what about

```
if households.empty:
    pop[copy_cols[col]] = pd.Series(np.nan, index=pop.index, dtype=object)
    continue
# lines 528-547 here
```

Reply:
I had tried something very similar to this with continue to break out of the loop for that column and I couldn't get it to work but I will keep trying. It seems like it should and I agree it is a lot cleaner.

Reply:
@mattkappel  if I use continue I still hit line 541 which is the original cause of the bugfix. I thought using continue would send me out of the for loop for the iteration and into it for the next iteration.